### PR TITLE
Enable soft bundle in devnet and testnet

### PR DESCRIPTION
**Pull request by shio-coder**
## Description 

This is a follow up change after PR #17508 and #17868 , for SIP-19.

The change adds a unit test for soft bundle failure modes, as well as enable soft bundle feature flag in devnet and testnet.

## Test plan 

The feature is not exactly user-facing and do not break anything.

---

## Release notes

- [x] Protocol: Enable soft bundle on Protocol Version 50 in devnet and testnet.
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Review by mystenmark — APPROVED** [review]
This all looks good. We need one additional test, which is to make sure that bad certs in soft bundles are rejected. The easiest way to do this is most likely to add another check here

https://github.com/MystenLabs/sui/blob/c9b0432a5ffead69977bd8a6843a3093a6cf9868/crates/sui-core/src/unit_tests/transaction_tests.rs#L435

which calls `handle_soft_bundle_certificates_v3` in addition to `handle_certificate_v2` and makes sure the error is the same. I think its okay to use a soft bundle of size one for this check.

**Comment by shio-coder** [comment]
Thanks Mark!  I have added forged cert tests for Soft Bundle.

It is a little bit different though because we have to make sure the tx has access to a shared object.
I have modified existing tests so that they run against two txs (one transfer, one move call w/ shared object).

File: crates/sui-protocol-config/src/lib.rs [hunk 1]
```diff
@@ -2589,6 +2595,14 @@ impl ProtocolConfig {
     pub fn set_mysticeti_num_leaders_per_round_for_testing(&mut self, val: Option<usize>) {
         self.feature_flags.mysticeti_num_leaders_per_round = val;
     }
+
+    pub fn set_enable_soft_bundle(&mut self, val: bool) {
+        self.feature_flags.soft_bundle = val;
+    }
+
+    pub fn set_max_soft_bundle_size(&mut self, val: u64) {
```
Comment (lxfind) [inline crates/sui-protocol-config/src/lib.rs:2603]: You shouldn't need this function, a macro is already generating test version of this function for all flags.

File: crates/sui-protocol-config/src/lib.rs [hunk 2]
```diff
@@ -2589,6 +2595,14 @@ impl ProtocolConfig {
     pub fn set_mysticeti_num_leaders_per_round_for_testing(&mut self, val: Option<usize>) {
         self.feature_flags.mysticeti_num_leaders_per_round = val;
     }
+
+    pub fn set_enable_soft_bundle(&mut self, val: bool) {
```
Comment (lxfind) [inline crates/sui-protocol-config/src/lib.rs:2599]: for this one, please add _for_testing suffix for consistency

**Review by lxfind — APPROVED** [review]

**Pull request merged**